plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -85,13 +96,54 @@ public ConnectorTableMetadata getTableMetadata()
     @Override
     public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
     {
-        if (commitInfoEntries.isEmpty()) {
-            return new FixedPageSource(ImmutableList.of());
+        Optional<Long> startVersion = Optional.empty();
+        Optional<Long> endVersion = Optional.empty();
+
+        if (constraint.getDomains().isPresent()) {
+            Map<Integer, Domain> domains = constraint.getDomains().get();
+            if (domains.containsKey(0)) {
+                Domain versionDomain = domains.get(0); // The zero value here relies on the column ordering defined in the constructor
+                Range range = versionDomain.getValues().getRanges().getSpan();
+                if (range.isSingleValue()) {
+                    long value = (long) range.getSingleValue();
+                    startVersion = Optional.of(value - 1);
+                    endVersion = Optional.of(value);
+                }
+                else {
+                    Optional<Long> lowValue = range.getLowValue().map(Long.class::cast);
+                    if (lowValue.isPresent()) {
+                        startVersion = Optional.of(lowValue.get() - (range.isLowInclusive() ? 1 : 0));
+                    }
+
+                    Optional<Long> highValue = range.getHighValue().map(Long.class::cast);
+                    if (highValue.isPresent()) {
+                        endVersion = Optional.of(highValue.get() - (range.isHighInclusive() ? 0 : 1));
```

Comment:
rename startVersion and endVersion to be either `*Inclusive` or `Exclusive` (startVersionExclusive and endVersionInclusive, right?)